src/tx/builder/helpers.ts

```diff
@@ -32,58 +32,63 @@ export const createSalt = salt
 
 export { encode, decode }
 
+export interface Pointer {
+  key: string
+  id: string
+}
+
 /**
  * Build a contract public key
- * @function
  * @alias module:@aeternity/aepp-sdk/es/tx/builder/helpers
- * @param {string} ownerId The public key of the owner account
- * @param {number} nonce the nonce of the transaction
- * @return {string} Contract public key
+ * @param ownerId The public key of the owner account
+ * @param nonce the nonce of the transaction
+ * @return Contract public key
  */
-export function buildContractId (ownerId, nonce) {
-  const ownerIdAndNonce = Buffer.from([...decode(ownerId, 'ak'), ...toBytes(nonce)])
+export function buildContractId (ownerId: string, nonce: number | BigNumber): string {
+  const ownerIdAndNonce = Buffer.from([...decode(ownerId as EncodedData<'ak'>), ...toBytes(nonce)])
   const b2bHash = hash(ownerIdAndNonce)
   return encode(b2bHash, 'ct')
 }
 
 /**
  * Build a oracle query id
- * @function
- * @function* @alias module:@aeternity/aepp-sdk/es/tx/builder/helpers
- * @param {String} senderId The public key of the sender account
- * @param {Number} nonce the nonce of the transaction
- * @param {Number} oracleId The oracle public key
- * @return {string} Contract public key
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder/helpers
+ * @param senderId The public key of the sender account
+ * @param nonce the nonce of the transaction
+ * @param oracleId The oracle public key
+ * @return Contract public key
  */
-export function oracleQueryId (senderId, nonce, oracleId) {
-  function _int32 (val) {
+export function oracleQueryId (
+  senderId: string,
+  nonce: number | BigNumber | string,
+  oracleId: EncodedData<'ok'>): string {
+  function _int32 (val: number): Buffer {
     const nonceBE = toBytes(val, true)
     return concatBuffers([Buffer.alloc(32 - nonceBE.length), nonceBE])
   }
 
-  const b2bHash = hash(Buffer.from([...decode(senderId, 'ak'), ..._int32(nonce), ...decode(oracleId, 'ok')]))
+  const b2bHash = hash(Buffer.from([...decode(senderId as EncodedData<'ak'>), ..._int32(nonce as number), ...decode(oracleId)]))
```

Comment:
```suggestion
  const b2bHash = hash(Buffer.from([...decode(senderId as EncodedData<'ak'>), ..._int32(nonce), ...decode(oracleId)]))
```
should we update `toBytes` definition instead?